src/app/frontend/download_data.py

```diff
@@ -0,0 +1,48 @@
+from typing import List
+
+import streamlit as st
+
+from genetic_testing.routers import ncbi
+
+
+def download_data(database: str, ids: List[int], search_term: str) -> None:
+    """
+    Download sequence data from NCBI's database using the given UIDs.
+
+    Parameters
+    ----------
+    database : str
+        The name of the NCBI database from which data will be fetched.
+
+    ids : List[int]
+        A list of UIDs representing the entries to be fetched from the database.
+
+    search_term : str
+        The search term used to filter the data to be fetched.
+
+    Returns
+    -------
+    None
+        This function does not return any value. The fetched data is directly written to the Streamlit app.
+
+    Notes
+    -----
+    This function uses the `fetch_data` function from the `genetic_testing.routers.ncbi` module to retrieve data
+    from the specified NCBI database based on the provided list of IDs. If the `ids` list is empty, the function writes "Filtered Data is empty" to the Streamlit app. Otherwise, it fetches the data, converts it to a string, and writes "Fetched Data Successfully!" to the Streamlit app, along with a download button for the data. The downloaded file will be named "data.fasta" by default.
+    """
+    if len(ids) == 0:
+        st.write("Filtered Data is empty")
+    else:
+        file_buffer = ncbi.fetch_data(database, ids)
+        file_str = file_buffer.getvalue()
+        if len(file_str) == 0:
+            st.write(
+                f"NCBI's '{database}' database has no data for '{search_term}' search term"
+            )
+        else:
+            st.write("Fetched Data Successfully!")
+            st.download_button(
+                label="Download",
+                data=file_str,
+                file_name=f"{database}_{search_term}.fasta",
```

Comment:
Last question, will the search term contains terms with complex characters and spaces? maybe adding a sanitizer for the `search_term` to create a file name to would be great. Lower case, no spaces for file system sake.

Reply:
The `search term` will not contain complex characters but it will contain spaces. Removing the spaces will make the name difficult to interpret. 
Currently this is how it look - 

![image](https://github.com/uw-ssec/neglected-diagnostics/assets/25303819/10ed5130-7b9a-40bd-961e-5e4676c1ab05)

The PIs will have the option to rename the file before they download, so I think we can keep it as it is? What do you think?

Reply:
I see. Yea that makes sense. Let's just keep it as it is.